sample_factory/algo/sampling/non_batched_sampling.py

```diff
@@ -125,11 +127,36 @@ def curr_actions(self) -> np.ndarray:
         :return: the latest set of actions for this actor, calculated by the policy worker for the last observation
         """
         actions = ensure_numpy_array(self.last_actions)
-        if self.env_info.integer_actions:
+
+        if isinstance(self.env_info.action_space, gym.spaces.Discrete):
+            return self.calculate_actions()
+
+        if isinstance(self.env_info.action_space, gym.spaces.Box):
+            return self.calculate_actions()
+
+
+        if isinstance(self.env_info.action_space, gym.spaces.Tuple):
+            # calculate splits
+            splits = [calc_num_actions(space) for space in self.env_info.action_space]
```

Comment:
I think this solution is good, except maybe I'd cache things like calc_num_actions in the ctor to avoid doing this every time.

Also, may I suggest an alternative solution?
What about if we have mixed actions we just leave them all to be float and the env can then convert some of the actions to ints that need to be converted?
The rationale is that the env knows which actions are which type anyway.

Since you're the author of the env with mixed actions, you're a better authority on this, I don't feel strongly either way :)